engine-precompiles/src/native.rs

```diff
@@ -689,4 +750,46 @@ mod tests {
     fn test_exit_with_valid_amount() {
         validate_amount(U256::from(u128::MAX)).unwrap();
     }
+
+    #[test]
+    fn test_parse_recipient() {
+        assert_eq!(
+            parse_recipient(b"test.near").unwrap(),
+            Recipient {
+                receiver_account_id: "test.near".parse().unwrap(),
+                message: None
+            }
+        );
+
+        assert_eq!(
+            parse_recipient(b"test.near:unwrap").unwrap(),
+            Recipient {
+                receiver_account_id: "test.near".parse().unwrap(),
+                message: Some("unwrap")
+            }
+        );
+
+        assert_eq!(
+            parse_recipient(b"test.near:some_msg:with_extra_colon").unwrap(),
+            Recipient {
+                receiver_account_id: "test.near".parse().unwrap(),
+                message: Some("some_msg:with_extra_colon")
+            }
+        );
+
+        assert_eq!(
+            parse_recipient(b"test.near:").unwrap(),
+            Recipient {
+                receiver_account_id: "test.near".parse().unwrap(),
+                message: Some("")
```

Comment:
Should this be `None`?

Reply:
I think it should be an empty string due to the colon at the end.